spyder/plugins/projects/plugin.py

```diff
@@ -422,10 +410,8 @@ def open_project(self, path=None, project=None, restart_consoles=True,
                 self.set_project_filenames(
                     self.main.editor.get_open_filenames())
 
-            # TODO: Don't emit sig_project_closed when we support
-            # multiple workspaces.
-            self.sig_project_closed.emit(
-                self.current_active_project.root_path)
+            # TODO: Don't stop watcher when we support multiple workspaces.
+            self.watcher.stop()

```

Comment:
This is wrong. We need to inform the LSP that the current project was closed and why we need to use `sig_project_closed` for that.

Reply:
Okay, I'll have to look into the emit. This was causing the documents in the editor for the first project to be reopened before opening the documents for the new project.